generators/lib/solution-explorer.js

```diff
@@ -0,0 +1,115 @@
+var traverse = require('traverse');
+var jsonpath = require('jsonpath');
+const tabdownParser = require('tabdown-kfatehi');
+const vsParser = require('./vs-parse/index');
+
+class SolutionFileExplorer {
+  constructor(content) {
+    this.content = content || {};
+    this.solution = {};
+  }
+
+  toString() {
+    return traverse(this.solution)
+      .reduce(function (acc, x) {
+        var hasValue = x && x.value;
+        var indent = (this.level - 2) / 2;
+        return hasValue && acc.push({ indent, value: x.value.trim() }), acc;
+      }, [])
+      .map((value) => `${'\t'.repeat(value.indent)}${value.value}`)
+      .join('\r\n');
+  }
+
+  findNode(keyword) { 
+    const result = jsonpath.query(this.solution, `$..children[?(@.value == "${keyword}" || @.value.match(/^.*${keyword}/g))]`);
+    if(result && result.length) {
+      return result[0];
+    }
+  }
+
+  insert(node, ...values) {
+    if (node) {
+      const index = this.solution.children.indexOf(node);
+      this.solution.children.splice(index + 1, 0, ...values);
+    }
+  }
+
+  getProjectNodes(projects) {
+    return projects
+    .map(project => [
```

Comment:
Oh. That looks very nice =) I have a feeling that now you can make it in 1 reduce call with a callback like "return [...acc,  firstLine, secondLine]"